cmake/Modules/Platform/Emscripten.cmake

```
@@ -87,33 +87,14 @@ else()
 endif()
 
 # Specify the compilers to use for C and C++
-if ("${CMAKE_C_COMPILER}" STREQUAL "")
-  set(CMAKE_C_COMPILER "${EMSCRIPTEN_ROOT_PATH}/emcc${EMCC_SUFFIX}" CACHE FILEPATH "Emscripten emcc")
-endif()
-if ("${CMAKE_CXX_COMPILER}" STREQUAL "")
-  set(CMAKE_CXX_COMPILER "${EMSCRIPTEN_ROOT_PATH}/em++${EMCC_SUFFIX}" CACHE FILEPATH "Emscripten em++")
-endif()
-
-if ("${CMAKE_AR}" STREQUAL "")
-  set(CMAKE_AR "${EMSCRIPTEN_ROOT_PATH}/emar${EMCC_SUFFIX}" CACHE FILEPATH "Emscripten ar")
-endif()
-
-if ("${CMAKE_RANLIB}" STREQUAL "")
-  set(CMAKE_RANLIB "${EMSCRIPTEN_ROOT_PATH}/emranlib${EMCC_SUFFIX}" CACHE FILEPATH "Emscripten ranlib")
-endif()
-
-if ("${CMAKE_C_COMPILER_AR}" STREQUAL "")
-  set(CMAKE_C_COMPILER_AR "${CMAKE_AR}" CACHE FILEPATH "Emscripten ar")
-endif()
-if ("${CMAKE_CXX_COMPILER_AR}" STREQUAL "")
-  set(CMAKE_CXX_COMPILER_AR "${CMAKE_AR}" CACHE FILEPATH "Emscripten ar")
-endif()
-if ("${CMAKE_C_COMPILER_RANLIB}" STREQUAL "")
-  set(CMAKE_C_COMPILER_RANLIB "${CMAKE_RANLIB}" CACHE FILEPATH "Emscripten ranlib")
-endif()
-if ("${CMAKE_CXX_COMPILER_RANLIB}" STREQUAL "")
-  set(CMAKE_CXX_COMPILER_RANLIB "${CMAKE_RANLIB}" CACHE FILEPATH "Emscripten ranlib")
-endif()
+set(CMAKE_C_COMPILER "${EMSCRIPTEN_ROOT_PATH}/emcc${EMCC_SUFFIX}" CACHE FILEPATH "Emscripten emcc")
```

```suggestion
set(CMAKE_C_COMPILER "${EMSCRIPTEN_ROOT_PATH}/emcc${EMCC_SUFFIX}" CACHE FILEPATH "Emscripten emcc" FORCE)
```

Without the `FORCE` I still have the same build failure like before ( as described here : https://github.com/emscripten-core/emscripten/issues/17992 ).

- [ ] TODO : Add `FORCE` (if possible)

---

Do you know why FORCE would be needed?  It seems a little odd and its not something I see in other toolchain files such as the NDK one linked above.

---

Adding `FORCE I believe would prevent overriding `CMAKE_C_COMPILER` on the command line or in an individual project.

---

Can you find out where `CMAKE_AR` is being set in your setup?

I think its a feature we want folks to be able override `CMAKE_AR` form, for example, the command line.